### PR TITLE
Use inline member initialization and default for ctor/dtors

### PR DESCRIPTION
**Pull request by orgads**

### CODE REVIEW

**Comment by jhasse** [comment]
Thanks, but we don't want to modernize the existing code right now.

**Comment by orgads** [comment]
Why not?

**Comment by jhasse** [comment]
Why would we? Any code change can be a regression and increases Git line history. So the question is never "why not?" but always "why?".

File: src/status.h [hunk 1]
```diff
@@ -88,7 +88,7 @@ struct StatusPrinter : Status {
   }
 
   struct SlidingRateInfo {
-    SlidingRateInfo(int n) : rate_(-1), N(n), last_update_(-1) {}
```
Comment (orgads) [inline src/status.h:91]: `rate` here is initialized with int, while it's a double.

**Review by orgads — COMMENTED** [review]
Found 1 missing initialization, and one with an implicit cast.

**Comment by orgads** [comment]
> > Prevents missed initializations
> 
> Theoretically or did you actually find one? This should be covered by unit tests + valgrind anyway.

Yes, I found one. That was my initiative for this change.

> Big con btw:
> 
> * Values are moved from cc to h so changes result in bigger recompiles.

Init values are hardly ever changed, so this shouldn't have any impact in real-life scenarios.

> Most of the changes to Ninja are bugfixes for real world issues where I (or the Owners) care.

Does anybody else maintain this project? You're the only one I find here.

**Pull request closed without merging**